packages/connectors/connector-dingtalk-web/src/index.ts

```diff
@@ -0,0 +1,162 @@
+} from './types.js';
+
+const getAuthorizationUri =
+  (getConfig: GetConnectorConfig): GetAuthorizationUri =>
+  async ({ state, redirectUri }) => {
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, dingtalkConfigGuard);
+
+    const { clientId, scope } = config;
+
+    const queryParameters = new URLSearchParams({
+      client_id: clientId,
+      redirect_uri: encodeURI(redirectUri), // The variable `redirectUri` should match {clientId, clientSecret}
+      response_type: 'code',
+      scope: scope ?? defaultScope,
+      state,
+      prompt: 'consent',
+    });
+
+    return `${authorizationEndpoint}?${queryParameters.toString()}`;
+  };
+
+export const getAccessToken = async (
+  code: string,
+  config: DingtalkConfig
+): Promise<{ accessToken: string }> => {
+  const { clientId, clientSecret } = config;
+
+  const httpResponse = await got.post(accessTokenEndpoint, {
+    json: {
+      clientId,
+      clientSecret,
+      code,
+      grantType: 'authorization_code',
+    },
+    timeout: { request: defaultTimeout },
+  });
+
+  const result = accessTokenResponseGuard.safeParse(parseJson(httpResponse.body));
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+  }
+
+  const { accessToken } = result.data;
+  assert(accessToken, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+
+  return { accessToken };
+};
+
+const getUserInfo =
+  (getConfig: GetConnectorConfig): GetUserInfo =>
+  async (data) => {
+    const { code } = await authorizationCallbackHandler(data);
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, dingtalkConfigGuard);
+    const { accessToken } = await getAccessToken(code, config);
+
+    try {
+      const httpResponse = await got.get(userInfoEndpoint, {
+        headers: {
+          'x-acs-dingtalk-access-token': accessToken,
+        },
+        timeout: { request: defaultTimeout },
+      });
+      const rawData = parseJson(httpResponse.body);
+      const result = userInfoResponseGuard.safeParse(rawData);
+
+      if (!result.success) {
+        throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+      }
+
+      const { nick: name, avatarUrl: avatar, unionId: id, email, mobile, stateCode } = result.data;
+      return {
+        id,
+        avatar,
+        phone: `${stateCode}${mobile}`,
+        email,
+        name,
+        rawData,
+      };
+    } catch (error: unknown) {
+      return getUserInfoErrorHandler(error);
+    }
+  };
+
+const getUserInfoErrorHandler = (error: unknown) => {
+  if (error instanceof HTTPError) {
+    const { statusCode, body: rawBody } = error.response;
+
+    if (statusCode === 401) {
+      throw new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid);
+    }
+
+    if (statusCode === 404) {
+      throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, 'user_not_found');
+    }
```

Comment:
Status code 404 should not lead to a invalid response error? better to attach doc link.

Reply:
Not sure whether a `/me` API can return 404.